Define ScyllaDB as drop-in replacement for Cassandra

mkorolyov opened the pull request:
## Which problem is this PR solving?

Defines that Jaeger server supports ScyllaDB as a storage backend as a drop-in replacement for Cassandra

## Short description of the changes
Updated README.md

mkorolyov commented:
i can provide a demo docker-compose file but have no idea where to place it to make it easily discoverable by the community

yurishkuro commented on README.md:
```diff
@@ -85,7 +85,8 @@ Jaeger can be used with a growing a number of storage backends:
 * It integrates via a gRPC API with other well known databases that have been certified to be Jaeger compliant: [TimescaleDB via Promscale](https://github.com/timescale/promscale), [ClickHouse](https://github.com/jaegertracing/jaeger-clickhouse).
 * There is embedded database support using [Badger](https://github.com/dgraph-io/badger) and simple in-memory storage for testing setups.
-* There are ongoing community experiments using other databases, such as ScyllaDB, InfluxDB, Amazon DynamoDB, YugabyteDB(YCQL).
+* It supports ScyllaDB as a drop-in replacement for Cassandra because ScyllaDB uses the same data model and query language as Cassandra.
```
```suggestion
* ScyllaDB can be used as a drop-in replacement for Cassandra since it uses the same data model and query language.
```

yurishkuro reviewed (commented):
We can add a folder plugins/storage/scylladb with a README and a docker-compose file. The README can call out any relevant details (or any differences vs. Cassandra). But I would like to have a clear disclaimer at the top that this is not an officially supported backend, meaning  that we will not proactively address any issues that may arise from incompatibilities between the two databases (we may still accept PRs).

yurishkuro commented:
Please make sure all commits are signed (or squash into one and sign).

yurishkuro reviewed (commented):
lgtm, but please make sure commits are signed (see CONTRIBUTING.md)

mkorolyov commented on plugin/storage/scylladb/README.md:
```diff
@@ -0,0 +1,40 @@
+
+Docker compose file consists of Jaeger server, Jaeger Cassandra schema writer, Jaeger UI, Jaeger Demo App `HotRod` and a ScyllaDB cluster.
+
+There is known issues with docker compose network configuration and containers connectivity on Apple Silicone. That's why before `upping` the docker compose you need to manually create network which later be used by docker compose:
```
I can't find the issue and also I wasn't able to reproduce it 100% times. I saw comments there and here about networking issues on macs and especially on m1. it works as expected on my second laptop on m1 and doesn't on the current. There is a useful network detail [here](https://docs.docker.com/network/bridge/#differences-between-user-defined-bridges-and-the-default-bridge). I actually tried just to define a custom network in the compose file itself but it also was not as stable as just creating a separate network by hands and then using it in the compose.

we can rewrite this sentence to smth like `most stable approach for archiving containers connectivity on macs can look like this` or smth similar. sorry for my poor english

mkorolyov commented on plugin/storage/scylladb/README.md:
```diff
@@ -0,0 +1,40 @@
+
+#### Protocol version
+
+Jaeger server detects Cassandra protocol version automatically. At the date of the demo with specified versions server detects that it connected via protocol version 3 while it is actually 4. This leads to warn log in cassandra-schema container:
```
i have tried both
```
      CASSANDRA_PROTOCOL_VERSION: 4
      CASSANDRA_VERSION: 4
```
but nothing helped. It depects the version from interaction with ScyllaDB

yurishkuro commented on plugin/storage/scylladb/scylladb_healthcheck.sh:
```diff
@@ -0,0 +1,15 @@
+
+# Compare the actual number of nodes with the expected number
+if test "$ACTUAL_NODES" -ne "$EXPECTED_NODES"; then
+  exit 1
```
```suggestion
  echo "cluster not ready, expecting ${EXPECTED_NODES} nodes, found ${ACTUAL_NODES} nodes"
  exit 1
```

yurishkuro commented on plugin/storage/scylladb/scylladb_healthcheck.sh:
```diff
@@ -0,0 +1,15 @@
+  exit 1
+fi
+
+exit 0
```
```suggestion
echo "cluster is ready with ${ACTUAL_NODES} nodes"
exit 0
```

yurishkuro commented on plugin/storage/scylladb/README.md:
````diff
@@ -0,0 +1,40 @@
+
+Create ScyllaDB cluster with 3 nodes and give it some time to start(around 1 minute):
+```shell
+docker compose up -d scylladb scylladb2 scylladb3
````
how about something like this? (or bettr a version that exist the loop on success)
```suggestion
docker compose up -d scylladb scylladb2 scylladb3
while sleep 5; do bash scylladb_healthcheck.sh; done
```

yurishkuro commented on plugin/storage/scylladb/README.md:
````diff
@@ -0,0 +1,40 @@
+```shell
+docker compose up -d
+```
+Open Demo app in your browser: http://localhost:8080 and make some clicks 
````
```suggestion
Open Demo app in your browser: http://localhost:8080 and click some buttons
```

yurishkuro commented on plugin/storage/scylladb/README.md:
```diff
@@ -0,0 +1,40 @@
+
+Docker compose file consists of Jaeger server, Jaeger Cassandra schema writer, Jaeger UI, Jaeger Demo App `HotRod` and a ScyllaDB cluster.
+
+There is known issues with docker compose network configuration and containers connectivity on Apple Silicone. That's why before `upping` the docker compose you need to manually create network which later be used by docker compose:
```
```suggestion
There is a known issue with docker compose network configuration and containers connectivity on Apple silicone. Sometimes it's helpful to manually create the docker network before running `docker compose up`:
```

yurishkuro reviewed (dismissed):
few small improvements

mkorolyov commented:
Hi @yurishkuro 
Thanks for the review. I reworked health checks and simplified things a lot. Also, I applied all the style nits you posted above.

I still use `docker compose up -d` instead of the suggested `docker compose up` because without background mode console is filled with tons of logs from 3 ScyllaDB nodes communicating with each other to gather into a single cluster and even after that it is not obvious that you can start using HotRod or JaegerUI. While with background mode it is transparent that ScyllaDB is starting for some time and then all other containers are spinning up and you can try JaegerUI or HotRod.

yurishkuro reviewed (approved):
Thanks. One possible minor improvement is to pull the Jaeger and ScyllaDB versions into variables at the top of the compose file.
